### PR TITLE
Script to automatically update customlists with search criteria

### PR DESCRIPTION
**Pull request by RishiDiwanTT**
## Description

Script to automatically update customlists with search criteria
    
Currently it only supports JSON type saved searches
with no support for additional facets
    
Pending items:
- Actual crontab entry

## Motivation and Context
The need for library administrators to save specific searches into lists and have them automatically update.
[Notion](https://www.notion.so/lyrasis/CM-Automatically-add-new-titles-to-list-when-they-match-search-parameters-0191dee472764129a08ee8c2781236fd?d=a9c05e8141eb4aed8d77bb0f4989fd5f)

## Checklist

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/core/test_scripts.py [hunk 1]
```diff
@@ -3092,6 +3098,68 @@ def test_do_run(self):
...
+        mock_parse.return_value.libraries = [self._default_library]
+        script.parse_command_line = mock_parse
+
+        with freeze_time("2022-01-01") as frozen_time:
+            script.run()
+
+        self._db.refresh(custom_list)
+        self._db.refresh(custom_list1)
+        assert len(custom_list.entries) == 1 + len(
+            self.populated_books
```
Comment (pcustic) [inline tests/core/test_scripts.py:3140]: This is not really related to this PR but why is `len(custom_list.entries) = 1 + ..`? Does this mean the empty list has one entry?

File: tests/core/test_scripts.py [hunk 2]
```diff
@@ -3092,6 +3098,68 @@ def test_do_run(self):
         assert 1 == customlist.size
 
 
+class TestCustomListUpdateEntriesScript(EndToEndSearchTest):
+    def populate_works(self):
+        self.populated_books: list[Work] = [
+            self._work(with_license_pool=True, title="Populated Book") for _ in range(5)
+        ]
+        self.unpopular_books: list[Work] = [
```
Comment (pcustic) [inline tests/core/test_scripts.py:3106]: Is this on purpose that one is popu**lated** and other unpopu**lar**? Should they be (`populated` and `unpopulated`) or (`popular` and `unpopular`) or this is on purpose? 😄

**Review by pcustic — APPROVED** [review]
This seems fine to me. I added two questions and one suggestion but actual code looks fine to me.

File: tests/core/test_scripts.py (hunk 2 above)
Comment (RishiDiwanTT) [reply tests/core/test_scripts.py:3106]: Yes, I was bored

File: tests/core/test_scripts.py (hunk 1 above)
Comment (RishiDiwanTT) [reply tests/core/test_scripts.py:3140]: The testing method _customlist() creates a list with one work in it already

File: core/scripts.py [hunk 3]
```diff
@@ -3494,6 +3496,85 @@ def do_run(self):
...
+            wl = WorkList()
+            wl.initialize(custom_list.library)
+            works = wl.search(
+                self._db, json_query, search, pagination=pagination, facets=facets
+            )
+
+            ## No more works
+            if not len(works):
+                self.log.info(
+                    f"Completed customlist auto update {custom_list.name} with {total_works_updated} works"
```
[the remark this answers is not in the document]
Comment (RishiDiwanTT) [reply core/scripts.py:3560]: This is a fair observation, I will make this change
